bin/process-csv.js

```diff
@@ -31,143 +31,163 @@
 
     // Generate "products.csv" based on the source data.
     processProducts = function (workPath, targetPath) {
-        var skippedRecords = [];
+        var skippedRecords = [],
+            authors = {};
         console.log('Processing CSV data with workPath="%s", targetPath="%s"', workPath, targetPath);
         fs.readdir(targetPath + '/images', function (err, imageFilenames) {
             if (err) {
                 console.error('Could not read images path "%s", please check the path exists and try again', targetPath + '/images');
                 return;
             }
-            fs.createReadStream(workPath + '/Books.csv')
+            fs.createReadStream(workPath + '/Authors.csv')
                 .pipe(csv.parse({ columns: true }))
-                .pipe(csv.transform(function (record) {
-                    var title, imageFilename;
-
-                    if (!record.RRP) {
-                        skippedRecords.push({ record: record, reason: 'RRP is zero or missing' });
-                        return undefined;
+                .pipe(csv.transform(function (author) {
+                    var authorName = (author.AuthFirstName + ' ' + author.AuthLastName).trim();
+                    if (authorName.length > 0) {
+                        authors[author.AuthorID] = authorName;
                     }
-
-                    title = record.Title.trim()
-                        .replace(/\s*[-(]\s*out\s+of\s+(?:stock|print|order).*$/i, '')
-                        .replace(/\s*[-(]\s*special\s+orders?\s+only.*$/i, '');
-
-                    imageFilename = _.find(imageFilenames, (filename) => filename.match(new RegExp('^' + record.BookCode + '_')));
-
-                    return {
-                        sku: 'BOOK-' + record.BookCode,
-                        store_view_code: '',
-                        attribute_set_code: 'Default',
-                        product_type: 'simple',
-                        categories: categoryMap[record.Category.trim()] || 'Default Category/Books/Other',
-                        product_websites: 'base',
-                        name: title,
-                        description: record.Description,
-                        short_description: '',
-                        weight: 1,
-                        product_online: 1,
-                        tax_class_name: 'Taxable Goods',
-                        visibility: 'Catalog, Search',
-                        price: record.RRP,
-                        special_price: '',
-                        special_price_from_date: '',
-                        special_price_to_date: '',
-                        url_key: slug(title) + '-' + record.BookCode,
-                        meta_title: title,
-                        meta_keywords: [ 'Western Australia', 'History', record.Category.toLowerCase() ].join(', '), // TODO Others? Author?
```

Comment:
This TODO is done, the author name is now used in both the `keywords` and `attributes` fields.